pkg/controller/acmeorders/controller.go

```diff
@@ -158,6 +158,8 @@ func NewController(
 }
 
 func (c *controller) ProcessItem(ctx context.Context, key string) error {
+	c.scheduledWorkQueue.Forget(key)
```

Comment:
Question: here and for the trigger controller's `ProcessItem` func- do we need to do this? I see no harm in doing it, however we will anyway always remove the key from queue when adding it as well. If there's a need to do this could we add a comment?

Reply:
Good point, this line isn't self-explanatory at all. I'll add a comment.

The purpose of `Forget` is to cancel the running timer when the Certificate gets deleted; otherwise, when a Certificate is deleted without calling `Forget`, the goroutine will keep running forever.

Ideally, the `Forget` call should only happen when the item gets deleted. We don't need to `Forget` on Certificate "Updated" and "Added".

Not sure how I can achieve that though. Maybe I can set up another informer.HandleEvent:

```go
issuerInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
    DeleteFunc: c.scheduledWorkQueue.Forget,
})
```

I'd definitely need to have a test case around that though

Reply:
I imagine for this PR just a comment may be sufficient. I imagine that it's not a major issue if we `Forget` on updates/adds too.

Reply:
I tried writing a test for it, but the whole `Register` + controller.Builder makes it really tricky to test :( 

I'll add a comment that explains why it exists and that this is untested.